### DISCUSSION TITLE
setup-homebrew action fails in self-hosted gha-runner-scale-set

### DISCUSSION THREAD

**Discussion by cbc02009** [Homebrew/discussions#4934] [Everyday usage] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

I am trying to use homebrew in a self-hosted runner to install a couple of tools but when it runs it always fails with the following error: 
```
Could not find 'brew' command in PATH or standard locations.
Error: The process '/usr/bin/bash' failed with exit code 1
```
The github action steps can be viewed [here](https://github.com/cbc02009/k8s-home-ops/blob/bde30e68751977560d2032b48f9694c9fa8ac4aa/.github/workflows/kubeconform.yaml#L35)

The full action log is available [here](https://github.com/cbc02009/k8s-home-ops/actions/runs/7035309496/job/19145382026)

The self-hosted runners are running in k8s on Talos Linux boxes.

The action runs fine when using `runs-on: ubuntu-latest` instead of the self-hosted runner.

Is there anything specific I need to setup in my action runner to allow for brew to work?

**Comment by ZhongRuoyu** [comment, marked as answer]
Is Homebrew installed in the runner? If not, you need to install it there first. The `setup-homebrew` action does not install Homebrew; it sets up the shell environment so that `brew` can be used in subsequent steps.

Homebrew is already installed in the `ubuntu-latest` runner image, so the action works there. ([It's just not in `PATH`.](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#homebrew-note))

**Reply by cbc02009** [reply]
Gotcha, I have not installed homebrew. I thought that's what the action did, my mistake. Is there an action to install homebrew, or do I just run a step with the shell commands?

**Reply by ZhongRuoyu** [reply]
There's a one-liner as documented in https://github.com/Homebrew/install (and on Homebrew's homepage https://brew.sh/):

```shell
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

To run it non-interactively, set the `NONINTERACTIVE` environment variable:

```shell
NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

If the runner is reused (i.e. not ephemeral), you probably only need to install it once; no need to have in your `action.yml`.

**Reply by cbc02009** [reply]
thanks, now I need to figure out what the base image for the gha action runner sets is and figure out how to install curl. At least I know what the issue is. Much appreciated.
